SpringProject/src/test/java/com/gradproject2019/UserServiceTest.java

```diff
@@ -15,4 +23,39 @@
 
     @Mock
     private UserRepository userRepository;
-}
+
+    private final User qwerty = new User( 3L, "qwerty", "qwerty", "qwerty", "qwerty@qwerty.com", "Qwerty!1", "qwerty");
+
+    @Test(expected = CredentialsExistException.class)
+    public void shouldThrowErrorWhenUsernameExists() {
+        //given user adds username
+        given(userRepository.findByUsername("qwerty")).willReturn(Optional.of(qwerty));
+        UserRequestDto copycat = createUserRequestDto("notqwerty","notqwerty", "notqwerty@qwerty.com", "notqwerty", "NotQwerty!1",  "qwerty");
+
+        //when username already exists
+        userService.saveUser(copycat);
+    }
+
+    @Test(expected = CredentialsExistException.class)
+    public void shouldThrowErrorWhenEmailExists() {
+        //given user adds email
+        given(userRepository.findByEmail("qwerty@qwerty.com")).willReturn(Optional.of(qwerty));
+        UserRequestDto copycat = createUserRequestDto("notqwerty","notqwerty", "qwerty@qwerty.com", "notqwerty", "NotQwerty!1",  "notqwerty");
```

Comment:
I see these magic strings are being used in more than one place, so they should be extracted into their own `private static` fields

Reply:
This has been amended